ml-agents/mlagents/trainers/tf_policy.py

```diff
@@ -290,13 +326,158 @@ def export_model(self):
             tf2bc.convert(frozen_graph_def_path, self.model_path + ".nn")
             logger.info("Exported " + self.model_path + ".nn file")
 
+        # logger.info("Converting to onnx")
+        self.convert_frozen_to_onnx(output_graph_def, ["vector_observation:0","epsilon:0"], ["action:0","action_probs:0"])
+
+    def convert_frozen_to_onnx(self, frozen_graph_def, inputs, outputs):
+        # This is bascailly https://github.com/onnx/tensorflow-onnx/blob/master/tf2onnx/convert.py
+        # but without the command line
+        from tf2onnx.tfonnx import process_tf_graph, tf_optimize
+        from tf2onnx import constants, loader, logging, utils, optimizer
+        from tf2onnx.loader import freeze_session, remove_redundant_inputs
+
+        # Save specific constants that we need to propagate, so that we can add them back later.
+        constant_names = {
+            "memory_size",
+            "version_number",
+            "is_continuous_control",
+            "action_output_shape",
+        }
+        constant_values = {}
+        for n in frozen_graph_def.node:
+            if n.name in constant_names:
+                val = n.attr["value"].tensor.int_val[0]
+                constant_values[n.name] = val
+
+        # args
+        fold_const = False
+        continue_on_error = False
+        target = []
+        opset = None
+        custom_ops = {}
+        extra_opset = []
+        shape_override = None
+        inputs_as_nchw = None
+
+        logging.basicConfig(level=logging.get_verbosity_level(1))
+
+        def make_constant_node_proto(name, value):
+            from onnx import (
+                TensorProto,
+                AttributeProto,
+                NodeProto,
+            )
+
+            # EXAMPLE
+            # node {
+            #     output: "truediv_3/y:0"
+            #     name: "truediv_3/y"
+            #     op_type: "Nop"
+            #     attribute {
+            #       name: "dtype"
+            #       i: 1
+            #       type: INT
+            #     }
+            #     attribute {
+            #       name: "value"
+            #       t {
+            #         data_type: 1
+            #         name: "truediv_3/y:0"
+            #         raw_data: "\000\000@@"
+            #       }
+            #       type: TENSOR
+            #     }
+            #     domain: "com.unity"
+            #   }
+            dtype_attribute = AttributeProto(
+                name="dtype", i=int(TensorProto.INT32), type=AttributeProto.INT
+            )
+            tensor_value = TensorProto(
+                data_type=TensorProto.INT32, name=name, int32_data=[value], dims=[1, 1, 1, 1]
+            )
+            value_attribute = AttributeProto(
+                name="value", t=tensor_value, type=AttributeProto.TENSOR
+            )
+            return NodeProto(
+                output=f"{name}:0",
+                name=name,
+                op_type="Constant",
```

Comment:
Though this was correct based on https://github.cds.internal.unity3d.com/unity/UnityInferenceEngine/blob/0378a31b135bdb1604b59d582a54f8c67869124c/UnityProject/Assets/Barracuda/Plugins/Editor/BarracudaEditor/ONNXModelImporter.cs#L58 but not seeing anything in the import